src/otaclient_common/retry_task_map.py

```diff
@@ -120,7 +128,10 @@ def _task_done_cb(
 
     def _fut_gen(self, interval: int) -> Generator[Future[Any], Any, None]:
         finished_tasks = 0
-        while True:
+        while finished_tasks == 0 or finished_tasks != self._total_task_num:
```

Comment:
finished_tasks == 0 is the initial status.
When finished_tasks become non-zero, it indicates that tasks execution is started, then we track finished_tasks != self._total_task_num to ensure all tasks are executed. Note that self._total_tasks_num only has three kinds of possible value, see L68-L75.